roqoqo/src/noise_models/overrotation.rs

```diff
@@ -0,0 +1,442 @@
+            gate: gate.to_string(),
+            theta_mean,
+            theta_std,
+        }
+    }
+
+    /// Return gate name of SingleQubitOverrotationDescription.
+    ///
+    /// # Returns
+    ///
+    /// `gate` - Returns gate name.
+    pub fn gate(&self) -> &String {
+        &self.gate
+    }
+
+    /// Return mean of Gaussian distrbution of overrotation angles name of SingleQubitOverrotationDescription.
+    ///
+    /// # Returns
+    ///
+    /// `theta_mean` - mean of distrbution.
+    pub fn theta_mean(&self) -> f64 {
+        self.theta_mean
+    }
+
+    /// Return standard deviation of Gaussian distrbution of overrotation angles name of SingleQubitOverrotationDescription.
+    ///
+    /// # Returns
+    ///
+    /// `theta_std` - standard deviation of distrbution.
+    pub fn theta_std(&self) -> f64 {
+        self.theta_std
+    }
+}
+
+/// Single qubit overrotation noise model on gate.
+///
+/// Adds a rotatation gate with a randomly distributed rotation angle after specified gates in a quantum circuit.
+/// Example:
+///
+/// ```
+/// use roqoqo::noise_models::SingleQubitOverrotationDescription;
+/// use roqoqo::noise_models::SingleQubitOverrotationOnGate;
+/// use roqoqo::noise_models::NoiseModel;
+/// let gate = "RotateX";
+/// let theta_mean = 0.0;
+/// let theta_std = 1.0;
+/// let mut noise_desc = SingleQubitOverrotationDescription::new(gate, theta_mean, theta_std);
+///
+/// let mut noise = SingleQubitOverrotationOnGate::new();
+/// let circuit_gate_with_noise = "RotateZ";
+/// let qubit = 0;
+/// noise = noise.set_single_qubit_overrotations(circuit_gate_with_noise, qubit, noise_desc);
+/// let noise_model = NoiseModel::SingleQubitOverrotationOnGate(noise);
+/// ```
+#[derive(Debug, Default, Clone, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(
+    feature = "serialize",
+    serde(from = "SingleQubitOverrotationOnGateSerialize")
+)]
+#[cfg_attr(
+    feature = "serialize",
+    serde(into = "SingleQubitOverrotationOnGateSerialize")
+)]
+pub struct SingleQubitOverrotationOnGate {
+    /// Overrotation noise information for single qubit gates in a quantum circuit.
+    single_qubit_overrotations: HashMap<(String, usize), SingleQubitOverrotationDescription>,
+    /// Overrotation noise information for two qubit gates in a quantum circuit.
+    two_qubit_overrotations: HashMap<
+        (String, (usize, usize)),
+        (
+            SingleQubitOverrotationDescription,
+            SingleQubitOverrotationDescription,
+        ),
+    >,
+}
+
+#[cfg(feature = "json_schema")]
+impl schemars::JsonSchema for SingleQubitOverrotationOnGate {
+    fn schema_name() -> String {
+        "SingleQubitOverrotationOnGate".to_string()
+    }
+
+    fn json_schema(gen: &mut schemars::gen::SchemaGenerator) -> schemars::schema::Schema {
+        <SingleQubitOverrotationOnGateSerialize>::json_schema(gen)
+    }
+}
+
+type SingleQGateIndex = (String, usize);
```

Comment:
I would suggest renaming this to SingleQubitGateIndex to match the two-qubit one